Assets/MixedRealityToolkit/_Core/Extensions/CollectionsExtensions.cs

```diff
@@ -113,5 +113,32 @@ public static void DisposeElements<TElement>(this IList<TElement> elements)
             }
         }
 
+        /// <summary>
+        /// Exports the values of a uint indexed Dictionary as an Array
+        /// </summary>
+        /// <typeparam name="T">Type of data stored in the values of the Dictionary</typeparam>
+        /// <param name="input">Dictionary to be exported</param>
+        /// <returns>array in the type of data stored in the Dictionary</returns>
+        public static T[] ExportDictionaryValuesAsArray<T>(this Dictionary<uint,T> input)
+        {
+            T[] output = new T[input.Count];
+            input.Values.CopyTo(output, 0);
+            return output;
+        }
+
+        /// <summary>
+        /// Overload extension to enable saving of an InteractionDefinition inside a Dictionary collection
+        /// *Note can only use generics (in both here and InteractionDefinition)
+        /// </summary>
+        /// <typeparam name="T">Type of input being saves</typeparam>
+        /// <param name="input">The InteractionDefinition dictionary reference (generics, performed on a Dictionary)</param>
+        /// <param name="key">The specific DeviceInputType value to update</param>
+        /// <param name="value">The data value to be updated</param>
+        public static void SetDictionaryValue<T>(this Dictionary<Definitions.Devices.DeviceInputType, Definitions.Devices.InteractionDefinition> input, Definitions.Devices.DeviceInputType key, T value)
```

Comment:
Could this be generic?

Reply:
Yes, because the values we set are generic and this now uses the generic SET method you created :D 
Else, I would need to create overloaded versions for each value type.